Assets/Plugins/CountlySDK/CountlyUtils.cs

```diff
@@ -50,12 +50,12 @@ public string GetUniqueDeviceId()
             foreach (var item in TimeMetricModel.GetTimeMetricModel())
                 baseParams.Add(item.Key, item.Value);
 
-            if (!string.IsNullOrEmpty(_countly.OptionalParameters.CountryCode))
-                baseParams.Add("country_code", _countly.OptionalParameters.CountryCode);
-            if (!string.IsNullOrEmpty(_countly.OptionalParameters.City))
-                baseParams.Add("city", _countly.OptionalParameters.City);
-            if (_countly.OptionalParameters.Location != null)
-                baseParams.Add("location", _countly.OptionalParameters.Location);
+            if (!string.IsNullOrEmpty(_countly.RecordLocation.CountryCode))
```

Comment:
Location information is not part of base params

Reply:
currently, location, country code and city send with each request.
location will send only with session begin?

Reply:
it should be according to the spec:
https://support.count.ly/hc/en-us/articles/360037753291-SDK-development-guide#recording-location

Reply:
location will send only with session begin.

Reply:
Make changes assuming that consent exists. By that logic your planned action is wrong.